### PR TITLE
Rewrite profile_pics to be generic

### PR DESCRIPTION
**Pull request by leadwolf**
- move `profile_pics` to `pics`
- for every item type (actors, scenes...) the plugin receives a configurable list of prop<>path entries to search in
- search is recursive via `$walk`, match is according to `$matcher`

### CODE REVIEW

**Comment by leadwolf** [comment]
Maybe wait until there is a $find equivalent of $walk ?

**Comment by leadwolf** [comment]
Updated $walk usage to prevent iterating when only need a single match. Results are acummulated outside the callback, allowing for multiple results

**Comment by leadwolf** [comment]
Todo: replace getAllExtra with maxExtra

File: plugins/pics/info.json [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    },
+    {
+      "name": "actors.[0].blacklistTerms",
+      "type": "string",
+      "required": false,
+      "default": [],
+      "description": "Terms that should not be found in an image path"
+    },
+    {
+      "name": "actors.[0].max ",
```
Comment (boi123212321) [inline plugins/pics/info.json:66]: Space

Comment (leadwolf) [reply plugins/pics/info.json:66]: I don't understand

Comment (boi123212321) [reply plugins/pics/info.json:66]: "actors.[0].max " should be "actors.[0].max"

Comment (boi123212321) [reply plugins/pics/info.json:66]:
```suggestion
      "name": "actors.[0].max",
```

I forgot I could do suggestions

Comment (leadwolf) [reply plugins/pics/info.json]: Oh ok, I thought I was crazy

**Comment by leadwolf** [comment]
adultempire failing, a title seems to have changed

**Pull request merged**